services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/DockerManagerImplUnitTest.java

```diff
@@ -307,7 +346,84 @@ public void testDockerRegistryNotificationPull() {
 				DockerRegistryEventUtil.createDockerRegistryEvent(RegistryEventAction.pull, REGISTRY_HOST, USER_ID, REPOSITORY_PATH, TAG, DIGEST);
 		dockerManager.dockerRegistryNotification(events);
 		// no create operation, since the repo already exists
-		verify(entityManager, never()).createEntity((UserInfo)any(), (Entity)any(), (String)any());
+		verify(entityManager, never()).createEntity((UserInfo)anyObject(), (Entity)anyObject(), anyString());
+		verify(dockerCommitDao, never()).createDockerCommit(anyString(), anyLong(), (DockerCommit)anyObject());
+	}
+	
+	private static DockerCommit createCommit() {
+		DockerCommit commit = new DockerCommit();
+		Date createdOn = new Date();
+		commit.setCreatedOn(createdOn);
+		commit.setDigest(DIGEST);
+		commit.setTag(TAG);
+		return commit;
+	}
+	
+	@Test
+	public void testAddDockerCommitToUnmanagedRespository() {
+		
+		DockerCommit commit = createCommit();
+		
+		// method under test
+		dockerManager.addDockerCommitToUnmanagedRespository(USER_INFO, REPO_ENTITY_ID, commit);
+		
+		verify(dockerCommitDao).createDockerCommit(REPO_ENTITY_ID, USER_ID, commit);
+		verify(transactionalMessenger).sendMessageAfterCommit(
+				eq(REPO_ENTITY_ID), eq(ObjectType.ENTITY), anyString(), eq(ChangeType.UPDATE));
+	}
+	
+	@Test(expected=IllegalArgumentException.class)
+	public void testAddDockerCommitToMANANGEDRespository() {
+		// this makes the repo a *managed* repo
+		when (dockerNodeDao.getRepositoryNameForEntityId(REPO_ENTITY_ID)).thenReturn(REPOSITORY_NAME);
+		DockerCommit commit = createCommit();
+
+		// method under test
+		dockerManager.addDockerCommitToUnmanagedRespository(USER_INFO, REPO_ENTITY_ID, commit);
+	}
+	
+	@Test(expected=UnauthorizedException.class)
+	public void testAddDockerCommitToUnmanagedRespositoryUNAUTHORIZED() {
+		
+		when(authorizationManager.canAccess(
+				eq(USER_INFO), eq(REPO_ENTITY_ID), eq(ObjectType.ENTITY), eq(ACCESS_TYPE.UPDATE))).
+				thenReturn(AuthorizationManagerUtil.ACCESS_DENIED);
+		
+		
+		DockerCommit commit = createCommit();
+		
+		// method under test
+		dockerManager.addDockerCommitToUnmanagedRespository(USER_INFO, REPO_ENTITY_ID, commit);
+	}
+	
+	@Test
+	public void listDockerCommits() {
+		List<DockerCommit> commits = new ArrayList<DockerCommit>();
+		commits.add(createCommit());
+		commits.add(createCommit());
+		
+		when(dockerCommitDao.listDockerCommits(REPO_ENTITY_ID, DockerCommitSortBy.CREATED_ON, /*ascending*/true, 10, 0)).
+			thenReturn(commits);
+		
+		when(dockerCommitDao.countDockerCommits(REPO_ENTITY_ID)).thenReturn(2L);
+		
+		// method under test
+		PaginatedResults<DockerCommit> pgs = dockerManager.listDockerCommits(
+				USER_INFO, REPO_ENTITY_ID, DockerCommitSortBy.CREATED_ON, /*ascending*/true, 10, 0);
+		
+		assertEquals(2L, pgs.getTotalNumberOfResults());
+		assertEquals(commits, pgs.getResults());
+	}
+	
```

Comment:
test that 4xx returns when user tries to list commits for an entity that is not a DockerEntity